src/instruction/jk_inst.rs

```diff
@@ -36,23 +55,56 @@ impl Instruction for JkInst {
     }
 
     fn print(&self) -> String {
-        match self {
-            JkInst::Dump => "@dump",
-            JkInst::Quit => "@quit",
+        match self.kind {
+            JkInstKind::Dump => "@dump",
+            JkInstKind::Quit => "@quit",
+            JkInstKind::Ir => "@ir",
         }
         .to_string()
     }
 
     fn execute(&self, interpreter: &mut Interpreter) -> Result<InstrKind, JkError> {
         interpreter.debug("JINKO_INST", &self.print());
 
-        match self {
-            JkInst::Dump => println!("{}", interpreter.print()),
-            JkInst::Quit => std::process::exit(0),
+        match self.kind {
+            JkInstKind::Dump => println!("{}", interpreter.print()),
+            JkInstKind::Quit => std::process::exit(0),
+            JkInstKind::Ir => eprintln!("usage: {:?} <statement|expr>", JkInstKind::Ir),
         };
 
         // JinkInsts cannot return anything. They simply act directly from the interpreter,
         // on the interpreter.
         Ok(InstrKind::Statement)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::parser::Construct;
+
+
+    #[test]
+    fn t_invalid_jkinst() {
+        let (_, fc) = Construct::function_call("tamer()").unwrap();
+        let inst = JkInst::from_function_call(fc);
+
+        assert!(inst.is_err(), "tamer is not a valid interpreter directive")
+    }
+
+    #[test]
+    fn t_valid_inst_no_args() {
+        let (_, fc) = Construct::function_call("dump()").unwrap();
+        let inst = JkInst::from_function_call(fc);
+
+        assert!(inst.is_ok(), "dump is a valid interpreter directive")
+    }
+
+    #[test]
+    fn t_valid_inst_with_args() {
+        let (_, fc) = Construct::function_call("ir(fn)").unwrap();
+        let inst = JkInst::from_function_call(fc);
+
+        assert!(inst.is_ok(), "ir(func) is a valid use of the ir interpreter directive")
+    }
```

Comment:
No test for ``dump(stuff_here)`` or ``ir()``

Reply:
They are both valid syntax and all interpreter directives are variadic, so I don't know if it should be tested.